pkg/crc/machine/kubeconfig.go

```diff
@@ -240,3 +247,34 @@ func updateServerDetailsToKubeConfig(input, output string) error {
 	}
 	return clientcmd.WriteToFile(*cfg, output)
 }
+
+func mergeKubeConfigFile(kubeConfigFile string) error {
```

Comment:
Wondering if `ModifyConfig` in `vendor/k8s.io/client-go/tools/clientcmd/config.go` is achieving the same?

Reply:
Following I tried but it is producing empty kubeconfig file at the end. I tried to find out if this is really used for merge the 2 different config but didn't get. Either I am doing something wrong or may be this is not the function for merging config (go-doc does specify it should do all but not sure how)
```
func mergeKubeConfigFile(kubeConfigFile string) error {
	_, kubeConf1, err := getGlobalKubeConfig()
	if err != nil {
		return err
	}
	kubeConf2, err := clientcmd.LoadFromFile(kubeConfigFile)
	if err != nil {
		return err
	}
	kubeConfigAccess1 := clientcmd.NewDefaultClientConfig(*kubeConf1, &clientcmd.ConfigOverrides{})
	return clientcmd.ModifyConfig(kubeConfigAccess1.ConfigAccess(), *kubeConf2, false)
}
```

Reply:
https://github.com/crc-org/crc/issues/3602 created followup